mesh/ncmesh.cpp

```diff
@@ -3398,6 +3472,20 @@ void NCMesh::FindVertexCousins(int elem, int local, Array<int> &cousins) const
 
 //// Coarse/fine transformations ///////////////////////////////////////////////
 
+bool NCMesh::PointMatrix::operator==(const PointMatrix &pm) const
+{
+   MFEM_ASSERT(np == pm.np, "");
```

Comment:
Shouldn't this return `false` instead of returning an error?

Reply:
No, in this case comparing points with incompatible dimensions is an internal error.